libs/features/jarvis-chat.js

```diff
@@ -0,0 +1,343 @@
+let chatInitialized = false;
+let loadScript;
+let loadStyle;
+
+const isSilentEvent = (data) => (data['event.workflow'] === 'init' && data['event.type'] === 'request')
+  || (data['event.workflow'] === 'Chat' && data['event.type'] === 'load' && data['event.subtype'] === 'window');
+
+const getBaseEvent = (data) => ({
+  chatConversationId: data['event.context_guid'],
+  chatConversationSessionId: data['event.session_guid'],
+  chatWorkflow: data['event.workflow'],
+  chatCategory: data['event.category'],
+  chatSubcategory: data['event.subcategory'],
+  chatType: data['event.type'],
+  chatSubtype: data['event.subtype'],
+  chatUserguid: data['event.user_guid'],
+  chatLanguage: data['event.language'],
+  chatPagename: data['event.pagename'],
+  chatVisitorguid: data['event.visitor_guid'],
+  chatUrl: data['event.url'],
+  chatOrgguid: data['event.org_guid'],
+  chatReferrer: data['event.referrer'],
+  chatClientId: data['source.client_id'],
+  chatSourceName: data['source.name'],
+  chatPlatform: data['source.platform'],
+  chatContentName: data['content.name'],
+  chatEnv: data['env.com.name'],
+});
+
+const setSophiaLayer = (data) => {
+  const sophiaData = {
+    variationId: data['exp.variation_id'],
+    actionBlockId: data['exp.action_block_id'],
+    campaignId: data['exp.campaign_id'],
+    containerId: data['exp.container_id'],
+    controlGroupId: data['exp.control_group_id'],
+    treatmentId: data['exp.treatment_id'],
+    surfaceId: data['exp.surface_id'],
+  };
+
+  Object.keys(sophiaData).forEach((key) => {
+    if (!sophiaData[key]) delete sophiaData[key];
+  });
```

Comment:
Is there a possibility of zero or empty string values that need to be preserved?
```suggestion
  Object.keys(sophiaData).forEach((key) => sophiaData[key] ?? delete sophiaData[key]);
```

Reply:
There's no exact information about the data types and values, but technically some properties can be `0` or empty string values.